pkg/rancher-components/src/components/BadgeState/BadgeState.spec.ts

```diff
@@ -0,0 +1,14 @@
+import { shallowMount } from '@vue/test-utils'
```

Comment:
I like the idea of keeping tests as close to the actual component implementation itself. I was thinking that we could do something similar with storybook as well if we think that would be beneficial. 

The tradeoff with the directory structure that I chose is that we kind of end up with a lot of ceremony around creating new components.. Vuetify does something similar.

Reply:
The problem may be due the use of a flat tree file structure in the components, that may end up in a never ending scrolling for each time.
We may consider grouping components for keep files together.